package/swupdate/0002-stream_interface-need-to-interlock-the-stream-wakeup.patch

```diff
@@ -0,0 +1,83 @@
+From 4a70d2c2c094158b9f3ae73f94ead1976dd82ed6 Mon Sep 17 00:00:00 2001
+From: Charles Hardin <charles.hardin@chargepoint.com>
+Date: Wed, 24 May 2023 11:48:12 -0700
+Subject: [PATCH 1/1] stream_interface: need to interlock the stream wakeup
+ condition
+
+Because the cond wait can be triggered in multiple passes, the
+condition wait needs to be paired with a condition. The indication
+of the wakeup should be a boolean so that the loop can continue
+until the wakeup event has occurred.
+
+Example pattern that is expected - indenting is for highlighting
+
+    pthread_mutex_lock();
+        while(condition_is_false)
+            pthread_cond_wait();
+    pthread_mutex_unlock();
+
+Signed-off-by: Charles Hardin <charles.hardin@chargepoint.com>
```

Comment:
You should probably say that I signed off on it too. I want my name in there ;)

Reply:
You'll have to talk to Charles about that; these are the upstreamed patches, sorry!

Reply:
you can add David to the file - it makes him feel happy :)